tests/unit/test_hardware.py

```diff
@@ -0,0 +1,61 @@
+"""
+..
+   test_hardware.py
+
+Unit tests for pyra.hardware.py.
+"""
+# local imports
+from pyra import hardware
+
+
+def test_update_cpu():
+    """Tests if update_cpu returns a float value"""
+    test_cpu_percent = hardware.update_cpu()
+    assert test_cpu_percent <= 100  # test if value is number with a max value of 100
+    assert test_cpu_percent >= 0  # test if value is number with a min value of 0
+
+
+def test_update_gpu():
+    """
+    # todo... this test needs to be defined
+    """
+    pass
```

Comment:
Why can't we test this?

Reply:
No gpu in github runners

Reply:
You should write the test and automatically skip it if there's no GPU present. Pytest supports a skip decorator thing for this purpose. Minimally you can just use an if statement in the function itself.